storage/store/inmemory/unsynchronized/collections.go

```diff
@@ -128,3 +130,19 @@ func (c *Collections) Remove(collID flow.Identifier) error {
 
 	return nil
 }
+
+// AddToBatch adds all the in-memory storages to the given batch.
+// It is used for the batching writes to the DB.
+func (c *Collections) AddToBatch(batch storage.ReaderBatchWriter) error {
+	writer := batch.Writer()
+
+	for _, coll := range c.collections {
+		light := coll.Light()
+		err := operation.UpsertCollection(writer, &light)
```

Comment:
This needs to do what `StoreLightAndIndexByTransaction` does. please double check the main implementations to see what each should be storing.

e.g.
https://github.com/onflow/flow-go/blob/master/module/state_synchronization/indexer/indexer_core.go#L150
https://github.com/onflow/flow-go/blob/master/storage/store/collections.go#L143